src/main/src/test/java/org/geoserver/config/util/XStreamPersisterTest.java

```diff
@@ -1585,6 +1588,30 @@ public synchronized String toString(Locale locale) {
         assertFalse(converter.canConvert(Object.class));
     }
 
+    @Test
+    public void testInternationalStringConverter() throws IOException, ClassNotFoundException {
+        final String message = "This is an unlocalized message";
+        final SimpleInternationalString toTest = new SimpleInternationalString(message);
+        final ByteArrayOutputStream out = new ByteArrayOutputStream();
+        final ObjectOutputStream objectOut = new ObjectOutputStream(out);
+        objectOut.writeObject(toTest);
+        objectOut.close();
+        XStreamPersister persister = new XStreamPersisterFactory().createXMLPersister();
+        Converter candidate =
+                persister
+                        .getXStream()
+                        .getConverterLookup()
+                        .lookupConverterForType(SimpleInternationalString.class);
+        assertTrue(candidate instanceof XStreamPersister.InternationalStringConverter);
+        assertEquals(message, toTest.toString());
```

Comment:
Doesn't have that match sense to test the equality between the String and the InternationalString value here.
What I would test instead is the byte output stream content that contains the expected string value after the persister.save method has been executed.